### DISCUSSION TITLE
Working with migrations when ts-node fails

### DISCUSSION THREAD

**Discussion by billomore** [mikro-orm/mikro-orm#5275] [Show and tell]
`ts-node` fails to parse the TS type aliases in my repo; known issue, [PR](https://github.com/TypeStrong/ts-node/pull/1585) opened from 2021. Which means I can't use CLI, specifically for the very much desirable migrations functionality. Bummer!

However, [tsx](https://github.com/privatenumber/tsx) runs (almost) great, so I decided to run the migrations with it. After reading some CLI source code, I came up with the following two files:
_migration-create.ts_
```typescript
import { Migrator } from "@mikro-orm/migrations";
import { MikroORM, PostgreSqlDriver } from "@mikro-orm/postgresql";
import myConfig from "./mikro-orm.config";

const orm = await MikroORM.init<PostgreSqlDriver>(myConfig);
const migrator = new Migrator(orm.em);
await migrator.createMigration();
process.exit(0);
```
-migration-up.ts_
```typescript
import { MikroORM, PostgreSqlDriver } from "@mikro-orm/postgresql";
import myConfig from "./mikro-orm.config";

const orm = await MikroORM.init<PostgreSqlDriver>(myConfig);

const migrator = orm.getMigrator();
await migrator.up();
process.exit(0);
```

and in my `package.json`'s `scripts:
```json
"migration:create": "tsx ./migration-create.ts",
"migration:up": "tsx ./migration-up.ts"
```

It could be probably done smarter, maybe even on the CLI level (by adding one more `bin` entry point), but this is a quick and dirty way to make it work.

**Comment by acelaya** [comment]
I'm using `tsx` for other purposes in the project, and I was looking for a way to use it with MikroORM migrations, instead of `ts-node`.

This worked perfectly!

**Reply by B4nan** [reply]
The problem with `tsx` is that it uses esbuild which does not support legacy decorators (and metadata reflection on them), so it cannot be used as a universal replacement for ts-node, this would work only in a few cases but would be otherwise breaking. I'll think about a way to opt-in to this, as people who use `EntitySchema` or ts-morph instead of reflect-metadata should be just fine with `tsx`.

**Reply by acelaya** [reply]
I do use `EntitySchema` instead of decorators.

**Reply by acelaya** [reply]
> I'll think about a way to opt-in to this

Perhaps the right long-term approach would be to move to new standard TS decorators, but I can imagine this is far from a simple thing to do, specially while trying to keep backwards compatibility.

**Reply by B4nan** [reply]
v7 will be ESM only, so yes, that is surely coming (but probably next year).
